core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -643,10 +643,12 @@ private[spark] class ExecutorAllocationManager(
     // Should be 0 when no stages are active.
     private val stageAttemptToNumRunningTask = new mutable.HashMap[StageAttempt, Int]
     private val stageAttemptToTaskIndices = new mutable.HashMap[StageAttempt, mutable.HashSet[Int]]
-    // Number of speculative tasks pending/running in each stageAttempt
-    private val stageAttemptToNumSpeculativeTasks = new mutable.HashMap[StageAttempt, Int]
-    // The speculative tasks started in each stageAttempt
+    // Number of speculative tasks running in each stageAttempt
+    // TODO(SPARK-14492): We simply need an Int for this.
     private val stageAttemptToSpeculativeTaskIndices =
+      new mutable.HashMap[StageAttempt, mutable.HashSet[Int]]()
+    // Number of speculative tasks pending in each stageAttempt
+    private val stageAttemptToUnsubmittedSpeculativeTasks =
```

Comment:
How about renaming to `stageAttemptToPendingSpeculativeTasks` and renaming `stageAttemptToSpeculativeTaskIndices` to `stageAttemptToRunningSpeculativeTaskIndices`?

Reply:
`stageAttemptToSpeculativeTaskIndices` keeps the same meaning as `stageAttemptToTaskIndicies`, I would prefer to keep it~

Reply:
btw, `stageAttemptToPendingSpeculativeTasks` is a good idea to replace `stageAttemptToUnsubmittedSpeculativeTasks`, I'll fix it soon